crates/sui-types/src/committee.rs

```diff
@@ -22,14 +23,15 @@ pub struct Committee {
     pub total_votes: StakeUnit,
     // Note: this is a derived structure, no need to store.
     #[serde(skip)]
-    pub expanded_keys: HashMap<AuthorityName, PublicKey>,
+    expanded_keys: HashMap<AuthorityName, PublicKey>,
 }
 
 impl Committee {
     pub fn new(epoch: EpochId, voting_rights: BTreeMap<AuthorityName, StakeUnit>) -> Self {
         let total_votes = voting_rights.iter().map(|(_, votes)| votes).sum();
         let expanded_keys: HashMap<_, _> = voting_rights
             .iter()
+            // TODO: How do we guarantee the unwrap is safe?
```

Comment:
Good question: depending on the signature scheme, indeed not every byte sequence can derive a valid public key (i.e., zero points. not in the curve, blocklisted weak keys etc).

How do we register AuthorityNames atm? maybe we have this check when the voting_rights is created, but if not we should just return a `Result<Self, Error>` here if it fails, to prevent a malicious validator from panicking the app.